### DISCUSSION TITLE
Register peripheral

### DISCUSSION THREAD

**Discussion by samuelh2005** [cc-tweaked/CC-Tweaked#754] [Q&A] [answered]
So i've created a block with a tile entity in my mod , Now how do i register it as a peripheral. and yes i've got the library installed.
The block dose appear ingame and has the standard tile entity methods when warped by a computer.

**Comment by samuelh2005** [comment]
Also my tile entity extends ```LockableTileEntity``` and implements ```ISidedInventory, ITickableTileEntity```
and the block for it extends ```HorizontalBlock```

**Comment by SquidDev** [comment, marked as answer]
You either need to register an [IPeripheralProvider] or expose an [IPeripheral] capability on the tile entity.

Best thin to do is look at some of the peripherals within CC.
 - https://github.com/SquidDev-CC/CC-Tweaked/blob/e48427dbbcb9c0c2211564733e69dd66135b2b25/src/main/java/dan200/computercraft/shared/peripheral/generic/GenericPeripheralProvider.java#L36
 - https://github.com/SquidDev-CC/CC-Tweaked/blob/e48427dbbcb9c0c2211564733e69dd66135b2b25/src/main/java/dan200/computercraft/shared/peripheral/diskdrive/TileDiskDrive.java#L526-L530

[IPeripheralProvider]: https://github.com/SquidDev-CC/CC-Tweaked/blob/mc-1.15.x/src/main/java/dan200/computercraft/api/peripheral/IPeripheralProvider.java
[IPeripheral]: https://github.com/SquidDev-CC/CC-Tweaked/blob/mc-1.15.x/src/main/java/dan200/computercraft/api/peripheral/IPeripheral.java
